lib/matplotlib/legend.py

```diff
@@ -1269,6 +1278,45 @@ def _find_best_position(self, width, height, renderer, consider=None):
     def contains(self, event):
         return self.legendPatch.contains(event)
 
+    def set_draggable(self, state, use_blit=False, update='loc'):
+        """
+        Enable or disable mouse dragging support of the legend.
+
+        Parameters
+        ----------
+        state : bool
+            ``True`` / ``False`` enables / disables mouse dragging.
+        use_blit : bool, optional
+            Blitting is an old and basic but fast composition method for
+            images. Usually, you won't need to set this. However, you may use
+            this at the cost of a reduced quality if you have performance
+            issues when dragging. See also :ref:`func-animation`.
+        update : ['loc' | 'bbox'], optional
+            The legend parameter to be changed when dragged:
+
+            - 'loc': update the *loc* parameter of the legend
+            - 'bbox': update the *bbox_to_anchor* parameter of the legend
+
+        Returns
+        -------
+        If *state* is ``True`` this returns the `~.DraggableLegend` helper
+        instance. Otherwise this returns ``None``.
+        """
+        if state:
+            if self._draggable is None:
+                self._draggable = DraggableLegend(self,
+                                                  use_blit,
+                                                  update=update)
+        else:
+            if self._draggable is not None:
+                self._draggable.disconnect()
+            self._draggable = None
+        return self._draggable
```

Comment:
👍 Got it.  This is what confused me - this seems good, though I haven't tested it, so I'll not formally review.  If it languishes, feel free to ping me and I'll try to test.